examples/libf3d/python/qml/minimal_qml.py

```diff
@@ -0,0 +1,321 @@
+        if self._renderer is not None:
+            self._renderer.queue_mouse_release(x, y, button, modifiers)
+
+    # Wheel
+    @Slot(int, int, int)
+    def mouseWheel(self, delta_x: int, delta_y: int, modifiers: int) -> None:
+        if self._renderer is not None:
+            self._renderer.queue_wheel(delta_x, delta_y, modifiers)
+
+    # Keyboard
+    @Slot(int, str, int)
+    def keyPress(self, key: int, text: str, modifiers: int) -> None:
+        if self._renderer is not None:
+            self._renderer.queue_key_press(key, text, modifiers)
+
+    @Slot(int, int)
+    def keyRelease(self, key: int, modifiers: int) -> None:
+        if self._renderer is not None:
+            self._renderer.queue_key_release(key, modifiers)
+
+
+class F3DRenderer(QQuickFramebufferObject.Renderer):
+    """
+    Actual OpenGL renderer for F3DView.
+    """
+
+    def __init__(self, item: F3DView):
+        super().__init__()
+        self.item = item
+
+        f3d.Engine.autoload_plugins()
+
+        self.engine = f3d.Engine.create_external(
+            QOpenGLContext.currentContext().getProcAddress
+        )
+
+        if self.item.modelPath:
+            self.engine.scene.add(self.item.modelPath)
+
+        win = self.engine.window
+        win.size = (int(self.item.width()), int(self.item.height()))
+
+        opt = self.engine.options
+        opt.update(
+            {
+                "render.grid.enable": True,
+                "render.show_edges": False,
+                "ui.axis": True,
+                "ui.fps": True,
+                "ui.filename": True,
+                "render.background.color": [0.15, 0.15, 0.15],
+            }
+        )
+
+        self.interactor = self.engine.interactor
+
+        self._events = deque()
+
+    def queue_mouse_press(self, x: float, y: float, button: int, modifiers: int):
+        self._events.append(("mouse_press", (x, y, button, modifiers)))
+        self.update()
+
+    def queue_mouse_move(self, x: float, y: float, buttons: int, modifiers: int):
+        self._events.append(("mouse_move", (x, y, buttons, modifiers)))
+        self.update()
+
+    def queue_mouse_release(self, x: float, y: float, button: int, modifiers: int):
+        self._events.append(("mouse_release", (x, y, button, modifiers)))
+        self.update()
+
+    def queue_wheel(self, dx: int, dy: int, modifiers: int):
+        self._events.append(("wheel", (dx, dy, modifiers)))
+        self.update()
+
+    def queue_key_press(self, key: int, text: str, modifiers: int):
+        self._events.append(("key_press", (key, text, modifiers)))
+        self.update()
+
+    def queue_key_release(self, key: int, modifiers: int):
+        self._events.append(("key_release", (key, modifiers)))
+        self.update()
+
+    # --------------------------
+    # Event mapping helpers
+    # --------------------------
+
+    def _update_modifiers(self, mods: int) -> None:
+        if self.interactor is None:
+            return
```

Comment:
you don't need these checks in this example because `self.interactor` is set in `__init__` and can't be `None`